kobo/apps/hook/models/hook_log.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+from importlib import import_module
+import logging
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext as _
+from jsonbfield.fields import JSONField as JSONBField
+from rest_framework import status
+from rest_framework.reverse import reverse
+import requests
+
+from ..constants import HOOK_LOG_PENDING, HOOK_LOG_FAILED, HOOK_LOG_SUCCESS
+from kpi.fields import KpiUidField
+
+
+class HookLog(models.Model):
+
+    hook = models.ForeignKey("Hook", related_name="logs", on_delete=models.CASCADE)
+    uid = KpiUidField(uid_prefix="hl")
+    instance_uuid = models.CharField(default="", max_length=36, db_index=True)  # `kc.logger.Instance.uuid`. Useful to retrieve data on retry
+    tries = models.PositiveSmallIntegerField(default=0)
+    status = models.PositiveSmallIntegerField(default=HOOK_LOG_PENDING)  # Could use status_code, but will speed-up queries.
+    status_code = models.IntegerField(default=200)
+    message = models.TextField(default="")
+    date_created = models.DateTimeField(auto_now_add=True)
+    date_modified = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        ordering = ["-date_created"]
+
+    def can_retry(self):
+        """
+        Returns whether instance can be resent to external endpoint.
+        Notice: even if returns false, `self.retry()` can be triggered.
+
+        :return: bool
+        """
+        if self.hook.active:
+            seconds = 60 * (10 ** settings.HOOK_MAX_RETRIES)  # Must match equation in `task.py:L60`
```

Comment:
Could this equation be moved into a `staticmethod` on `Hook` or `HookLog`? Then that could be called here and from:
* https://github.com/kobotoolbox/kpi/pull/1864/files#diff-3a1cd180601a48439bab076b7e2ef0a6R33
* https://github.com/kobotoolbox/kpi/pull/1864/files#diff-0e490e23f4fba4abcfb40faca624dc7bR173

(not blocking)

Reply:
Actually I found another bug thanks to your observation. 
It should not be retried until all retries are done by Celery. So number of seconds is the sum of all seconds between each retry.

Fixed with https://github.com/kobotoolbox/kpi/commit/f95065e067d22a52482c937ab6dad9fdbb312ecd